05/app.js

```diff
@@ -0,0 +1,93 @@
+const REQUIRED_ERROR_MESSAGE = "To pole jest wymagane";
+const NUMBER_ERROR_MESSAGE = "To pole jest wymagane";
+const POSITIVE_NUMBER_ERROR_MESSAGE = "Podana wartość powinna być dodatnia";
+const ZIP_CODE_ERROR_MESSAGE = "Wprowadź poprawny kod pocztowy np.\"52-690\"";
+
+const errorMessages = [];
+
+document.addEventListener('DOMContentLoaded', init);
+function init() {
+    // console.log('.init()');
+
+    const formEl = document.querySelector('form');
+    formEl.noValidate = true;
+    formEl.addEventListener('submit', validateAll);
+}
+
+function validateAll(event) {
+    event.preventDefault();
+    let errors = [];
+    const formEl = event.currentTarget;
+    isRequired(formEl, errors);
+    isPositiveNumber(formEl, errors);
+    isMatching(formEl, errors);
+    colorErrors(errors);
+    addToMessages();
+}
+
+function addToMessages() {
+    errorMessages.forEach(function(message) {
+        const liEl = document.createElement('li');
+        liEl.innerText = message;
+        document.querySelector('.messages').appendChild(liEl);
+        //wiem, że to jest brzydko, bo zamiast wszystko na raz wcisnąć to ciągle się odwołuję, ale już nie mam pomysłu. (albo sił
+    })
+    errorMessages.length = 0;
+}
+
+function colorErrors(errors) {
+    errors.forEach(function(name) {
+        element = document.querySelector(`[name=${name}`);
+        if(element) {
+            element.style.border = '2px solid red';
+        }
+    });
+    errors.length = 0;
+}
+
+function isRequired(formEl, errors) {
```

Comment:
Osobiście w tej metodzie przekazywałbym tylko wartość konkretnego pola.
Wtedy to rozwiązanie będzie bardziej uniwersalne.
Teraz mocno wiążemy się z formularzem i sposobem oznaczenia pola wymaganego.
Wyszukanie elementu i dodanie komunikatu błędu bym zrobił poza tą funkcją.
Oczywiście nic nie stoi na drodze, aby zrobić to w innej funkcji.